tests/tests_views.py

```diff
@@ -0,0 +1,160 @@
+import pytest
+from users.models import Profile, Account_type, Meeting_method
+from django.test import Client
+from django.contrib.auth.models import User
+from users.forms import (RegistrationForm, ProfileForm, UserUpdateForm, ProfileUpdateForm)
+from django.urls import reverse
+
+USERNAME = "testuser"
+FIRSTNAME = "Test"
+LASTNAME = "User"
+PASSWORD = "testpass"
+EMAIL = "testuser@gmail.com"
+
+
+def create_test_user():
+    return User.objects.create_user(
+        username=USERNAME,
+        email=EMAIL,
+        password=PASSWORD
+    )
+
+
+@pytest.fixture
+def client():
+    return Client()
+
+
+@pytest.fixture
+def authenticated_client(client, new_user):
+    new_user.save()
+    client.login(username=USERNAME, password=PASSWORD)
+    return client
+
+
+@pytest.fixture
+def valid_registration_data():
+    return {
+        'username': 'test_user',
+        'email': 'test_user@test.com',
+        'password1': 'test_password',
+        'password2': 'test_password',
+        'account_type': Account_type.TEACHER,
+        'phone_number': '+1234567890',
+        'city': 'Test City',
+        'meeting_method': Meeting_method.LIVE
+    }
+
+
+@pytest.fixture
+def valid_profile_update_data():
+    return {
+        'bio': 'Test Bio',
+        'profession': 'Test Profession',
+        'phone_number': '+1234567890',
+        'city': 'Test City',
+        'account_type': Account_type.TEACHER,
+        'meeting_method': Meeting_method.LIVE
+    }
+
+
+@pytest.mark.django_db
+class TestProfilePageView:
+
+    def test_profile_page_authorized(self, authorized_client, persist_user_profile):
+        course_response = authorized_client.get('/users/'+persist_user_profile.user.username, follow=True)
+        assert course_response.status_code == 200
+        assert 'profile_details.html' in course_response.templates[0].name
+
+    def test_profile_page_unauthorized(self, client, persist_user_profile):
+        course_response = client.get('/users/'+persist_user_profile.user.username, follow=True)
+        assert course_response.status_code == 200
+        assert 'main/login.html' in course_response.templates[0].name
+
+
+@pytest.mark.django_db
+class TestRegester:
+
+    def test_regester_page(self, client):
+        profile_response = client.get('/register', follow=True)
+        assert profile_response.status_code == 200
+        assert 'register.html' in profile_response.templates[0].name
+
+    def test_RegistrationForm_valid_save(self, valid_registration_data):
+        form = RegistrationForm(data=valid_registration_data)
+        assert form.is_valid()
+        user = form.save()
+        assert user is not None
+
+    def test_ProfileForm_valid_save(self, valid_profile_update_data):
```

Comment:
Please follow PEP8 for naming conventions. Also, why do you need to check this form like that? sending the data on the page isn't enough?

Reply:
deleted

Reply:
Looks the same

Reply:
now it's ok